include/wil/coroutine.h

```diff
@@ -646,9 +683,27 @@ namespace wil::details::coro
 #define __WIL_COROUTINE_NON_AGILE_INCLUDED
 #include <ctxtcall.h>
 #include <wil/com.h>
+#include <roerrorapi.h>

```

Comment:
Is this generally applicable if COM headers are included? I've always associated the restricted error info more with WinRT than with COM. Just curious if it makes sense to factor this out into its own `ifdef` if `roerrorapi.h` has already been included (assuming a unique include guard exists for it).

Reply:
I factored it out into a separate bock with suitable include guards but it is disabled in our app because it seems that <roerrorapi.h> is not included by anything else that we use.

That separation would vastly limit the usefulness of this error propagation because none of the VS project templates would include it by default.  But even if it isn't included COM is still providing restricted error information so it really needs to be propagated.

In case it is interesting I see that WRL headers include it so there is definitely precedent for including this header broadly.

Reply:
WRL was mostly created for the Windows Runtime (although you can of course use it to consume/produce non-WinRT types), so not necessarily a precedent setter here.

That said, I don't have many strong feelings here. I think it's ultimately fine either way.